### DISCUSSION TITLE
feature request: add option to prevent teams installation

### DISCUSSION THREAD

**Discussion by soredake** [farag2/Sophia-Script-for-Windows#492] [Ideas]
### Prerequisites

- [X] Verify that this is not a Windows issue;
- [X] Refer to the [system requirements](https://github.com/farag2/Sophia-Script-for-Windows#system-requirements);
- [X] Refer to the [How to use](https://github.com/farag2/Sophia-Script-for-Windows#how-to-use);
- [X] I do not use a homebrew Windows image;
- [X] I did not tweak Windows before that could cause system instability;
- [ ] If your issue concerns the Wrapper, please mention @BenchTweakGaming in issue

### Steps to reproduce

It would be nice if sophiscript had option to prevent teams installation https://woshub.com/remove-teams-chat-windows/ https://www.outsidethebox.ms/21375/

The problem is "ConfigureChatAutoInstall" registry key needs to be added with TrustedInstaller rights.

### Windows Version

Windows 11 Pro

### Sophia Script version

6.5.0

**Comment by farag2** [comment]
Что, реально от TI надо создать ключ? :)

**Reply by soredake** [reply]
yes
![image](https://github.com/farag2/Sophia-Script-for-Windows/assets/5204968/3fe78f94-45a5-44b0-98bc-caae12eaf63e)

**Reply by farag2** [reply]
Проблема не в реализации, в необходимости. Я выложу готовую функцию вечерком, но добавлять такое полотно кода я не буду в основную ветку. :)
Жди...
По-английски я уже устал писать за день. Сорянствую.

**Comment by farag2** [comment]
В чем проблема просто скрыть значок, наплевав на то, что он триггерит скачивание? Нет значка — нет проблемы. Просто там только получение прав 250 строк C#-кода.

**Reply by soredake** [reply]
Меня напрягает что оно устанавливается автоматически, если ты не хочешь это добавлять можно просто закрыть это как wontfix.

**Reply by farag2** [reply]
То, что не буду добавлять, это и ежу уже ясно, ибо крайне специфическая хрень, в которой я не особо шарю. Я — про изменения прав через консоль. Но сам код я выложу. Позже...
Просто забавный запрос. Такое надо делать на стадии создания образа, а не онлайн. Короче, посмотрим...

**Comment by farag2** [comment]
```powershell
$Task = @"
New-ItemProperty -Path HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Communications -Name ConfigureChatAutoInstall -Value 0 -Type Dword -Force
"@
$Parameters = @{
	TaskName = "RunAsTI"
	Action   = New-ScheduledTaskAction -Execute powershell.exe -Argument "-WindowStyle Hidden -Command $Task"
}
Register-ScheduledTask @Parameters -Force

$ScheduleService = New-Object -ComObject Schedule.Service
$ScheduleService.Connect()
$ScheduleService.GetFolder("\").GetTask("RunAsTI").RunEx($null, 0, 0, "NT SERVICE\TrustedInstaller')
```

Проще, чем это, не найти. Полотна на 700 строк я не сомг осилить. Элегантно, однако. Ты убедила: я добавлю эту балалайку в 6.5.1 :)
